c_src/kos_msg_nif.c

```diff
@@ -499,6 +516,27 @@ static ERL_NIF_TERM n_kos_dir_request_str(ErlNifEnv* env, int argc, const ERL_NI
       payload_term));
 }
 
+static ERL_NIF_TERM n_kos_status_atom_map(ErlNifEnv* env, int _argc, const ERL_NIF_TERM _argv[]) {
+  // If we don't construct a new NIF here we get a segfault.
+  ERL_NIF_TERM map;
+  
+#define STATUS_DECL(k, v) enif_make_atom(env, #k) ,
```

Comment:
That's a nice way of binding the two values together and then expanding them out into separate arrays.